src/Providers/Http/DTO/Request.php

```diff
@@ -0,0 +1,280 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Providers\Http\DTO;
+
+use InvalidArgumentException;
+use WordPress\AiClient\Common\AbstractDataTransferObject;
+use WordPress\AiClient\Providers\Http\Enums\HttpMethodEnum;
+
+/**
+ * Represents an HTTP request.
+ *
+ * This class encapsulates HTTP request data that can be converted
+ * to PSR-7 requests by the HTTP transporter.
+ *
+ * @since n.e.x.t
+ *
+ * @phpstan-type RequestArrayShape array{
+ *     method: string,
+ *     uri: string,
+ *     headers: array<string, string|list<string>>,
+ *     body?: string|null
+ * }
+ *
+ * @extends AbstractDataTransferObject<RequestArrayShape>
+ */
+class Request extends AbstractDataTransferObject
+{
+    public const KEY_METHOD = 'method';
+    public const KEY_URI = 'uri';
+    public const KEY_HEADERS = 'headers';
+    public const KEY_BODY = 'body';
+
+    /**
+     * @var HttpMethodEnum The HTTP method.
+     */
+    protected HttpMethodEnum $method;
+
+    /**
+     * @var string The request URI.
+     */
+    protected string $uri;
+
+    /**
+     * @var array<string, string|list<string>> The request headers.
+     */
+    protected array $headers;
+
+    /**
+     * @var string|null The request body.
+     */
+    protected ?string $body;
+
+    /**
+     * Constructor.
+     *
+     * @since n.e.x.t
+     *
+     * @param HttpMethodEnum|string $method The HTTP method.
+     * @param string $uri The request URI.
+     * @param array<string, string|list<string>> $headers The request headers.
+     * @param string|null $body The request body.
+     *
+     * @throws InvalidArgumentException If the URI is empty or method is invalid.
+     */
+    public function __construct($method, string $uri, array $headers = [], ?string $body = null)
+    {
+        if (empty($uri)) {
+            throw new InvalidArgumentException('URI cannot be empty.');
+        }
+
+        if (is_string($method)) {
+            $this->method = HttpMethodEnum::from(strtoupper($method));
+        } elseif ($method instanceof HttpMethodEnum) {
+            $this->method = $method;
+        } else {
+            throw new InvalidArgumentException('Method must be a string or HttpMethodEnum instance.');
+        }
+
+        $this->uri = $uri;
+        $this->headers = $headers;
+        $this->body = $body;
+    }
+
+    /**
+     * Gets the HTTP method.
+     *
+     * @since n.e.x.t
+     *
+     * @return string The HTTP method.
+     */
+    public function getMethod(): string
+    {
+        return $this->method->value;
+    }
+
+    /**
+     * Gets the HTTP method enum.
+     *
+     * @since n.e.x.t
+     *
+     * @return HttpMethodEnum The HTTP method enum.
+     */
+    public function getMethodEnum(): HttpMethodEnum
+    {
+        return $this->method;
+    }
```

Comment:
See above, this is inconsistent with our other DTOs: There needs to be only a single method `getMethod(): HttpMethodEnum`.

Reply:
Resolved in f2552c5